test/rdoc/test_rdoc_parser_prism_ruby.rb

```diff
@@ -1549,6 +1638,37 @@ class Bar; end
     assert_equal 'Foo::C', klass.find_module_named('C').full_name
   end
 
+  def test_constant_with_singleton_class
+    pend if accept_legacy_bug?
```

Comment:
If we don't want to run this test eternally, `omit` is better. `pend` is for a test that will be fixed later. 

```suggestion
    omit if accept_legacy_bug?
```

Reply:
Thank you. updated